### PR TITLE
URLs fix for production issues #187080649

### PR DESCRIPTION
**Pull request by tofarr**
This PR should:
* Fix the issue where an intake can be broken by hitting the wrong url.
* Fix the broken `/pending-federal-return` page
* Make the `/pending-federal-return` and `/canceled-data-transfer` state specific matching what we actually sent to directfile:
* Add a temporary redirect so that if direct file don't update their URL, users still get the correct message
image

You can test these by going to:
* `/en/ny/questions/pending-federal-return`
* `/en/ny/questions/canceled-data-transfer`

### CODE REVIEW

File: app/controllers/state_file/questions/canceled_data_transfer_controller.rb [hunk 1]
```diff
@@ -25,11 +25,6 @@ def illustration_path
 
       private
 
-      # this page sometimes doesn't have us_state in the params so can't use the default current_intake
-      def current_intake
-        GlobalID.find(session[:state_file_intake])
-      end
-
```
Comment (tofarr) [inline app/controllers/state_file/questions/canceled_data_transfer_controller.rb:32]: This was wrong - intakes are are NEVER stored without devise now

File: app/controllers/state_file/questions/pending_federal_return_controller.rb [hunk 2]
```diff
@@ -4,13 +4,6 @@ class PendingFederalReturnController < QuestionsController
       def prev_path
         nil
       end
-
-      private
-
-      # edit page has no us_state in the params so default implementation of current_intake doesn't work
-      def current_intake
-        GlobalID.find(session[:state_file_intake])
-      end
```
Comment (tofarr) [inline app/controllers/state_file/questions/pending_federal_return_controller.rb:13]: Same as above

File: app/controllers/state_file/questions/return_status_controller.rb [hunk 3]
```diff
@@ -117,6 +119,21 @@ def reject_description
 
       def card_postscript; end
 
+      def redirect_if_no_submission
+        if current_intake.efile_submissions.empty?
+          redirect_to StateFile::Questions::InitiateDataTransferController.to_path_helper(us_state: state_code)
+        end
+      end
```
Comment (tofarr) [inline app/controllers/state_file/questions/return_status_controller.rb:126]: Without this, an end use can break their whole intake by simply visiting the /return-status URL before it is ready.

File: app/views/state_file/state_file_pages/fake_direct_file_transfer_page.html.erb [hunk 4]
```diff
@@ -36,15 +36,15 @@
   <%=
     link_to(
       "Cancel",
-      state_file_questions_canceled_data_transfer_path,
+      StateFile::Questions::CanceledDataTransferController.to_path_helper(us_state: current_intake.state_code),
```
Comment (tofarr) [inline app/views/state_file/state_file_pages/fake_direct_file_transfer_page.html.erb:39]: I had to change the format of this because these controllers are mapped differently

File: config/routes.rb [hunk 5]
```diff
@@ -549,8 +549,6 @@ def scoped_navigation_routes(context, navigation)
         namespace :questions do
           get "show_xml", to: "confirmation#show_xml"
           get "explain_calculations", to: "confirmation#explain_calculations"
-          get "pending_federal_return", to: "pending_federal_return#edit"
-          get "canceled_data_transfer", to: "canceled_data_transfer#edit"
```
Comment (mpidcock) [inline config/routes.rb:553]: Why are these removed?

**Review by mpidcock — APPROVED** [review]

File: config/routes.rb (hunk 5 above)
Comment (mpidcock) [reply config/routes.rb:553]: A: because this section is not state specific, the addition below is to "replace" these lines
